### DISCUSSION TITLE
Is pinia getter appropriate as a watch target???

### DISCUSSION THREAD

**Discussion by taylous** [vuejs/pinia#2407] [Help and Questions] [answered]
i'm not yet familiar with using Vue3.

i have been using it continuously for the past month.
one thing **i realized is that primitive variables or objects must be declared as refs to maintain responsiveness.**
> i understood that this is why storeToRef exists.

i know that the pinia getter caches values and has a readonly feature similar to Vue2's computed.

**so, is it not appropriate to use it on a watch?**
the reason I thought this was because I thought it would involve unnecessary calculations.
> When the getter returns the state value,

any help would be appreciated.

**Comment by posva** [comment, marked as answer]
it's okay to use a getter in a watcher:

```ts
watch(() => store.double, () => {})
```

Stores are reactive objects, so using a function is needed

- https://vuejs.org/guide/essentials/watchers.html#basic-example

**Reply by taylous** [reply]
sorry for the late response...
